OrbitGl/App.cpp

```diff
@@ -843,38 +843,41 @@ void OrbitApp::LoadSymbols(const std::filesystem::path& symbols_path,
                            const std::shared_ptr<Process>& process,
                            const std::shared_ptr<Module>& module,
                            const std::shared_ptr<PresetFile>& preset) {
-  // TODO (159889010) Move symbol loading off the main thread.
-  const auto scoped_status = CreateScopedStatus(absl::StrFormat(
+  auto scoped_status = CreateScopedStatus(absl::StrFormat(
       R"(Loading symbols for "%s" from file "%s"...)", module->m_FullName, symbols_path.string()));
-
-  const auto symbols_result = SymbolHelper::LoadSymbolsFromFile(symbols_path);
-  CHECK(symbols_result);
-  const ModuleSymbols& symbols = symbols_result.value();
-
-  module->LoadSymbols(symbols);
-  CHECK(process != nullptr);
-  process->AddFunctions(module->m_Pdb->GetFunctions());
-  LOG("Loaded %lu function symbols for module \"%s\"", symbols.symbol_infos().size(),
-      module->m_FullName);
-
-  // Applying preset
-  if (preset != nullptr) {
-    auto it = preset->preset_info().path_to_module().find(module->m_FullName);
-    if (it != preset->preset_info().path_to_module().end()) {
-      for (const FunctionInfo* func : module->m_Pdb->GetSelectedFunctionsFromPreset(*preset)) {
-        SelectFunction(*func);
+  thread_pool_->Schedule([this, scoped_status = std::move(scoped_status), symbols_path, process,
+                          module, preset]() mutable {
+    auto symbols_result = SymbolHelper::LoadSymbolsFromFile(symbols_path);
+    CHECK(symbols_result);
+    main_thread_executor_->Schedule([this, symbols = std::move(symbols_result.value()),
+                                     scoped_status = std::move(scoped_status), process, module,
+                                     preset] {
+      module->LoadSymbols(symbols);
+      CHECK(process != nullptr);
+      process->AddFunctions(module->m_Pdb->GetFunctions());
+      LOG("Loaded %lu function symbols for module \"%s\"", symbols.symbol_infos().size(),
+          module->m_FullName);
+
+      // Applying preset
+      if (preset != nullptr) {
+        auto it = preset->preset_info().path_to_module().find(module->m_FullName);
+        if (it != preset->preset_info().path_to_module().end()) {
+          for (const FunctionInfo* func : module->m_Pdb->GetSelectedFunctionsFromPreset(*preset)) {
+            SelectFunction(*func);
```

Comment:
I believe, selecting functions require to be done on the main thread.

Reply:
it is done on the main thread - the only part which is done of the main thread is line 850

Reply:
all legacy structures - like Module and Process are also modified on the main thread. Only loading from the elf-file has moved to thread-pool.

Reply:
Ah thanks. I missed line 852. Perfect. Then it seems to be all fine for me.